Changes requested in issue #112.

Matskuu opened the pull request:
Changes to generic achievement scalarizing functions requested in issue #112: add_asf_generic_diff and add_asf_generic_nondiff now take a second reference point and second weights to be used in the augmentation term. If a second reference point is not given, no reference point is used in the augmentation term. If second weights are not given, the same weights are used in both the max term and the augmentation term.

gialmisi commented on tests/test_scalarization.py:
Please modify the test at least so that you check that you get a different solution when you provide `reference_point` and `weights` only, and when you also provide the weights and reference point for the augmentation term.

Modify also the test function ` test_add_asf_generic_nondiff` with the same idea.

gialmisi reviewed (changes requested):
This seems to be fine. I would like the tests to be updated also before accepting this PR.

Matskuu commented:
I have now updated the test functions for the asf's. The differences for the nondiff version seem to be so small in the test case that I modified the rho value and only compared the decision variable values in the nondiff version testing.

gialmisi reviewed (approved):
These are fine now. The point of the tests, especially when solved using scipy, is to check that the scalarizations at least work without throwing any errors. So this is enough for now!
